corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js

```diff
@@ -185,29 +194,31 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
         }
     };
 
-    Utils.getCurrentQueryInputs = function () {
-        var queryData = Utils.currentUrlToObject().queryData[sessionStorage.queryKey];
-        if (queryData) {
-            return queryData.inputs || {};
-        }
-        return {};
+    Utils.getQueryInputs = function () {
+        return this.queryInputs || {};
     };
 
-    Utils.getStickyQueryInputs = function () {
-        if (!toggles.toggleEnabled('WEBAPPS_STICKY_SEARCH')) {
-            return {};
-        }
-        if (!this.stickyQueryInputs) {
-            return {};
+    Utils.setQueryInputs = function (inputs, queryKey) {
+        const queryInputs = Utils.getQueryInputs();
+        queryKey = queryKey || sessionStorage.queryKey;
+        if (queryKey && queryKey !== "null" && queryKey !== "undefined") {
+            queryInputs[queryKey] = inputs;
+            this.queryInputs = queryInputs;
```

Comment:
nit: can this just be `this.queryInputs[queryKey] = inputs`

Reply:
Good call, though I do at least have to check that `this.queryInputs` isn't undefined to set a key on it. What would you think about
```javascript
            this.queryInputs = this.queryInputs || {};
            this.queryInputs[queryKey] = inputs;
```
I could also call `getQueryInputs()` which is more DRY, but I think it's more efficient just to assign it to itself (or `{}` ) than assigning the result of a function.

Reply:
ah good catch for checking if `this.queryInputs` is undefined. Change look good